### PR TITLE
Changes to work with Otoole v1.0.3

### PR DESCRIPTION
**Pull request by EmiFej**
Changes include:

- Creation of a config.yaml file which is placed under the config folder. It includes the YearlyPeak and ReserveMarginTagTechnology for each country.

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def sol_gurobi(lp_path: str, environment, threads: int):
 
     return m
 
-def get_duals(model):
+#def get_duals(model):
```
Comment (willu47) [inline run.py:20]: Don't comment out the function definition here. It will create broken code and a potential source of bugs.

File: run.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def get_duals(model):
                 dic_duals[c] = pd.DataFrame(columns=['value', 'constraint', 'set_0', 'set_1', 'set_2'])
     return dic_duals
 
-def write_duals(dict_duals: dict, path: str):
+#def write_duals(dict_duals: dict, path: str):
```
Comment (willu47) [inline run.py:46]: Don't comment this out.

File: run.py [hunk 3]
```diff
@@ -72,12 +72,12 @@ def write_sol(sol, path_out: str, path_gen: str):
     lp_path = snakemake.input[0]
     outpath = snakemake.output[0]
     log_path = snakemake.log[0]
-    dual_path = snakemake.output[1]
+    #dual_path = snakemake.output[1]
```
Comment (willu47) [inline run.py:75]: No need to comment this.

File: run.py [hunk 4]
```diff
@@ -72,12 +72,12 @@ def write_sol(sol, path_out: str, path_gen: str):
...
     threads = snakemake.threads
 
     env = gp.Env(log_path)
 
     model = sol_gurobi(lp_path, env, threads)
-    dic_duals = get_duals(model)
-    write_duals(dic_duals, dual_path)
+    #dic_duals = get_duals(model)
+    #write_duals(dic_duals, dual_path)
```
Comment (willu47) [inline run.py:82]: These two lines are the only parts you must comment to deactivate writing the dual writing.  I guess a bug in the dual-value extraction prevents the solution file from being generated.

**Review by willu47 — CHANGES_REQUESTED** [review]
Hi @EmiFej. Good progress, but you need to find the source of the bug in `run.py` to close this off. See comments below.

File: run.py (hunk 4 above)
Comment (EmiFej) [reply run.py:82]: Please see the latest commit @willu47 1d84c3d. Let me know if the changes are in line with what was requested form the comments in this review.

File: snakefile [hunk 5]
```diff
@@ -22,8 +22,8 @@ rule convert_dp:
     conda:
         "envs/otoole_env.yaml"
     shell:
-        "otoole convert csv datafile {input.dp_path} {output.df_path}"
-
+        "otoole convert csv datafile {input.dp_path} {output.df_path} config/config.yaml"
+        #"otoole convert csv datafile {input.dp_path} {output.df_path}"
```
Comment (willu47) [inline snakefile:26]: Please remove this line 26

File: snakefile [hunk 6]
```diff
@@ -46,14 +46,14 @@ rule run_model:
         "working_directory/{scen}.lp",
     output:
          temp("working_directory/{scen}.sol"),
-         directory("working_directory/{scen}_duals")
+         #directory("working_directory/{scen}_duals")
```
Comment (willu47) [inline snakefile:49]: Add a comment explaining why this is commented out.

File: run.py (hunk 3 above)
Comment (willu47) [reply run.py:75]: Still to do

File: run.py (hunk 2 above)
Comment (willu47) [reply run.py:46]: Uncomment line 46

File: run.py (hunk 1 above)
Comment (willu47) [reply run.py:20]: Uncomment line 20

**Review by willu47 — CHANGES_REQUESTED** [review]
You still need to uncomment the function def lines on 20 and 26.

**Comment by EmiFej** [comment]
I'm not sure that everything is correct with my branch EmiFej-new-otoole-version. Some of the changes that you commented regarding the run.py file were already unhashed in my local version, but appeared as hashed out in the PR. Hopefully it is better now. Regarding the dual values I had issues with them before, hence the removal of all dual_value related parts of the code. I'm not sure if it was a code issue or on my end.

**Comment by EmiFej** [comment]
In commit d431128 it stated that `src/` was not staged or committed. After staging, committing, and pushing it to origin and changing the branch from `new-otoole-version` to `main` I got the following message:

`warning: unable to rmdir 'src/openentrance': Directory not empty
Switched to branch 'main'
Your branch and 'origin/main' have diverged,
and have 1 and 7 different commits each, respectively.`

I'm not sure that the file should be there or have the name` src/openentrance`.

**Pull request merged**